core/src/main/java/org/frankframework/pipes/EscapePipe.java

```diff
@@ -72,26 +72,34 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 			throw new PipeRunException(this, "cannot open stream", e);
 		}
 
-		String substring = null;
-		String result = input;
+		String result = handleSubstrings(input);
+		return new PipeRunResult(getSuccessForward(), result);
+	}
+
+	private String handleSubstrings(String input) {
+		if (substringStart == null || substringEnd == null) {
+			return handle(input);
+		}
+
+		StringBuilder result = new StringBuilder(input);
 		int i = -1;
-		int j = -1;
-		log.debug("substringStart [{}] substringEnd [{}] input [{}]", substringStart, substringEnd, input);
-		if (substringStart != null && substringEnd != null) {
-			i = input.indexOf(substringStart);
-			if (i != -1) {
-				j = input.indexOf(substringEnd, i);
-				if (j != -1) {
-					substring = input.substring(i + substringStart.length(), j);
-					substring = handle(substring);
-					result = input.substring(0, i + substringStart.length()) + substring + input.substring(j);
+		while ((i = result.indexOf(substringStart, i + 1)) != -1) {
+			int j = result.indexOf(substringEnd, i);
+			if (j != -1) {
+				try {
+					// Get the content between the substrings
+					String content = result.substring(i + substringStart.length(), j);
+
+					// Replace the new content in the original string
+					result.replace(i + substringStart.length(), j, handle(content));
+				} catch (NullPointerException nullPointerException) {
+					return substringStart + "null" + substringEnd;
```

Comment:
As discussed in Teams: catching a NPE should not be needed, but prevented. Please figure out where in the logic you can make an improvement so it won't happen.